### PR TITLE
build image based on current code

### PR DESCRIPTION
**Pull request by otherview**
Enables building the docker image base on the current code instead of the main github branch

### CODE REVIEW

File: dockerfiles/enclave.Dockerfile
Comment (tudor-malene) [inline dockerfiles/enclave.Dockerfile:3]: I don't know much about docker. Will this work on CI?

Comment (otherview) [reply dockerfiles/enclave.Dockerfile:3]: Yes it should work. Good thing you mentioned it, I did forget to update the CI cmd 👍

Comment (joeldudley) [reply dockerfiles/enclave.Dockerfile:3]: @otherview But this won't work in an environment where you don't have the repo cloned down, e.g. in the Azure VM.

File: dockerfiles/README.md
Comment (joeldudley) [inline dockerfiles/README.md:19]: @otherview Delete this extra line?

File: dockerfiles/README.md
Comment (joeldudley) [inline dockerfiles/README.md:21]: @otherview Capital 'O' in Obscuro for consistency.

File: dockerfiles/enclave.Dockerfile
Comment (otherview) [reply dockerfiles/enclave.Dockerfile:3]: Ah! I get what you mean 😃 

Typically, envs that only require the container (which is totally fair case) should download the built image from obscuro official dockerhup repository. If a build from scratch is needed, then the whole repo is required. 


I like nginx's example :  [dockerfile](https://github.com/nginxinc/docker-nginx/blob/6f0396c1e06837672698bc97865ffcea9dc841d5/mainline/debian/Dockerfile)  and https://hub.docker.com/_/nginx

File: dockerfiles/README.md
Comment (otherview) [reply dockerfiles/README.md:19]: Done 👍

File: dockerfiles/README.md
Comment (otherview) [reply dockerfiles/README.md:21]: Done 👍

File: dockerfiles/enclave.Dockerfile
Comment (otherview) [reply dockerfiles/enclave.Dockerfile:3]: Geth [Dockerfile](https://github.com/ethereum/go-ethereum/blob/master/Dockerfile) has a nice example I guess

Comment (joeldudley) [reply dockerfiles/enclave.Dockerfile:3]: I agree that’s the norm, but we need something that works until we have a pipeline to publish an image to DockerHub. This is going to break the Azure Deployer tool, for example.

Comment (tudor-malene) [reply dockerfiles/enclave.Dockerfile:3]: that's what I actually meant with my question as well.

Is there a way to have two dockerfiles? One that uses the current folder and one that does a checkout?
Or even better, write some logic like: "If this is running in an obscuro folder, do nothing, otherwise download from github?"

Comment (otherview) [reply dockerfiles/enclave.Dockerfile:3]: > I agree that’s the norm, but we need something that works until we have a pipeline to publish an image to DockerHub. This is going to break the Azure Deployer tool, for example.

Yes, I get the issue.

Should this be on hold until we have a pipeline that pushes images to dockerhub ?
I can build that pipeline, just need the green light for it :) 

The Azure Deployer tool seems to clone the repository and then build.
I'm probably missing something here aren't I?

https://github.com/obscuronet/obscuro-playground/blob/3f46f8209ff47326a44f236a97a09d08a62d6fd7/tools/azuredeployer/main/azure_deployer.go#L39-L44

> that's what I actually meant with my question as well.
> Is there a way to have two dockerfiles? One that uses the current folder and one that does a checkout?
Or even better, write some logic like: "If this is running in an obscuro folder, do nothing, otherwise download from github?"

Yeah that's a good idea. 
I'll leave the existing dockerfile untouched and move these changes to a new dockerfile ? 
Then CI can run on the new one 👍

**Review by tudor-malene — COMMENTED** [review]
looks good to me, but will defer to Joel, as I don't know anything about docker

**Review by joeldudley — APPROVED** [review]
LGTM. Sorry, I wasn't aware of the Docker `COPY` command. I thought it was a regular filesystem copy, and was confused at how files were expected to make it from the host to the container. But that's exactly what `COPY` does 🤦

**Pull request merged**
